### PR TITLE
Fixes #453 ink strokes not captured until stylus up

### LINKED ISSUE #453: Board with touchscreen not drawing as intended
Hello,

I was just trying out the board function of your software and noticed something weird. When I draw something on the board using the mouse, i can actually see the drawing of the single strokes in the resulting gif.

If I use my touch monitor to actually draw on it, the resulting gif will have the completed strokes pop in as one. Also the last few strokes seem to be missing for some reason.

I've attached two gifs to this issue. Hi_Mouse.gif is me drawing via the mouse. H_Touch.gif is me drawing via the touch monitor.

I was using:
ScreenToGif v2.17
Windows 10 Pro x64
HP E230t touch monitor

Attachments:
hi_mouse
hi_touch

### PR DESCRIPTION
**Pull request by theJosher**
This fixes #453 by using reflection to get a frozen copy of DrawingGroups from the inking thread's real-time visual. Under normal use cases, it seems strokes are transitioned from the inking thread to the UI thread's InkCanvas when completed (e.g. stylus up). However, in our sort of unique case of animating the drawing process, we do want to see the real-time visual incrementally, so this fix pulls over the partial stroke as DrawingGroups into an image overlay that is Z-ordered above the InkCanvas. This extra partial-stroke-copy logic is only performed when a capture is desired (via thetimer), and the inking overlay is emptied when no partial stroke is alive.

### CODE REVIEW

File: ScreenToGif/ImageUtil/ImageMethods.cs [hunk 1]
```diff
@@ -1504,6 +1504,33 @@ public static Icon ToIcon(this ImageSource imageSource)
             }
         }
 
+        /// <summary>
+        /// Visits all the descendents of the visual using DFS and adds frozen copies 
+        /// of their drawing objects as children to the drawingGroup argument.
+        /// </summary>
+        /// <param name="visual">The visual to convert to a DrawingGroup</param>
+        /// <param name="drawingGroup">The target DrawingGroup to be populated</param>
+        static public void visualToFrozenDrawingGroup(this Visual visual, DrawingGroup drawingGroup)
```
Comment (NickeManarin) [inline ScreenToGif/ImageUtil/ImageMethods.cs]: Hi, just a minor tweak in here, but I can do later:

`public static void VisualToFrozenDrawingGroup`

Comment (theJosher) [reply ScreenToGif/ImageUtil/ImageMethods.cs]: Should be updated now. Sorry, we have an awful coding standard at 🏭 to which I am strongly habituated. I think I fixed the flicker as well.

**Comment by NickeManarin** [comment]
Thanks for taking your time to deal with that issue. I'm waiting to get in hands with a computer with touch this week in order to test it.

**Comment by NickeManarin** [comment]
I still could not test it, is it working ok for you?

**Comment by theJosher** [comment]
Sorry I'm lame - I got pulled into overtime at work. There is still the
edge case that results in it going all wonky. I'll try to fix it this
weekend.

**Comment by NickeManarin** [comment]
Sorry for the delay again. I was able to test the recorder by using an input injector (mouse input to touch or pen input).

The recorder looks fine.

If it's still janky, maybe the `TransparentCornerPoint` could use the width and height of the current board size.
